flutter/news/lib/core/result.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:equatable/equatable.dart';
+
+abstract class Failure implements Exception {
+  final String message;
+  const Failure(this.message);
+}
+
+class Success<S> extends Equatable {
+  final S data;
+  const Success({required this.data});
+
+  @override
+  List<S> get props => [data];
+}
+
+class Result<S> extends Equatable {
+  final Failure? _failure;
+  final S? _data;
+
+  const Result(this._data, this._failure);
+
+  factory Result.failure(Failure failure) => Result(null, failure);
+  factory Result.success(S data) => Result(data, null);
+
+  Failure get failure {
+    assert(_failure != null);
```

Comment:
can you tell me why the "_" before the variable name ?
im not familiar with using the "_"

Reply:
hey @Diogo00dev , good question

The usage of `_` is to make the field private, as `dart` doesn't have `public`, `private`, `protected` keywords you prefix a field with `_` to make it private.

https://www.woolha.com/tutorials/dart-using-access-modifiers-private-public